tests/validator/validator.cpp

```diff
@@ -2348,7 +2420,47 @@ TEST(Validator, unfoundUnitsInEncapsulatedComponents)
     model->linkUnits();
     v->validateModel(model);
 
-    EXPECT_EQ_ERRORS(expectedErrors, v);
+    EXPECT_EQ_ISSUES(expectedIssues, v);
+}
+
+TEST(Validator, mismatchedMultipliersInUnits)
+{
+    const std::vector<std::string> expectedIssues = {
+        "Variable 'v1' in component 'Brobdingnag' has units of 'megametre' and an equivalent variable 'v2' in component 'Lilliput' has compatible but not equivalent units of 'millimetre'. The mismatch is: multiplication factor of 10^9."};
+    auto model = libcellml::Model::create("Gulliver");
+    auto c1 = libcellml::Component::create("Brobdingnag");
+    auto c2 = libcellml::Component::create("Lilliput");
+    auto v1 = libcellml::Variable::create("v1");
+    auto v2 = libcellml::Variable::create("v2");
+
+    auto u1 = libcellml::Units::create("megametre");
+    u1->addUnit("metre", "mega");
+
+    auto u2 = libcellml::Units::create("millimetre");
+    u2->addUnit("metre", "milli");
+
+    v1->setUnits(u1);
+    v2->setUnits(u2);
+    v1->setInterfaceType("public");
+    v2->setInterfaceType("public");
+    c1->addVariable(v1);
+    c2->addVariable(v2);
+    model->addComponent(c1);
+    model->addComponent(c2);
+    model->addUnits(u1);
+    model->addUnits(u2);
+
+    libcellml::Variable::addEquivalence(v1, v2);
+
+    auto validator = libcellml::Validator::create();
+    validator->validateModel(model);
+
+    EXPECT_EQ(size_t(1), validator->issueCount());
+    EXPECT_EQ(size_t(0), validator->errorCount());
+    EXPECT_EQ(size_t(1), validator->warningCount());
+    EXPECT_EQ(size_t(0), validator->hintCount());
+
+    EXPECT_EQ_ISSUES(expectedIssues, validator);
 }
```

Comment:
Why do we have this new test?

Reply:
We needed one to show that the valid model (ie: only mismatched multiplier, no errors) can still have a warning retrieved from it.

Reply:
Shouldn't that be done after this PR, i.e. once we have replaced our `Error` class with the `Issue` class, we can then add warnings/hints here and there?